5calls/app/src/main/java/org/a5calls/android/a5calls/model/AccountManager.java

```diff
@@ -0,0 +1,94 @@
+package org.a5calls.android.a5calls.model;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import org.a5calls.android.a5calls.FiveCallsApplication;
+
+/**
+ * A singleton responsible for managing all local account-related information
+ */
+public enum AccountManager {
+    Instance;
+
+    private static final String PREFS_FILE = "fiveCallsPrefs";
+
+    private static final String KEY_INITIALIZED = "prefsKeyInitialized";
+    private static final String KEY_ALLOW_ANALYTICS = "prefsKeyAllowAnalytics";
+    private static final String KEY_USER_ZIP = "prefsKeyUserZip";
+    private static final String KEY_LATITUDE = "prefsKeyLatitude";
+    private static final String KEY_LONGITUDE = "prefsKeyLongitude";
+
+    public boolean hasLocation(Context context) {
+        return (getLat(context) != null && getLng(context) != null) || !TextUtils.isEmpty(getZip(context));
+    }
+
+    // Defaults to true
+    public boolean isFirstTimeInApp(Context context) {
+        return getSharedPrefs(context).getBoolean(KEY_INITIALIZED, true);
+    }
+
+    public void setIsFirstTimeInApp(Context context, boolean isFirstTime) {
+        getSharedPrefs(context).edit().putBoolean(KEY_INITIALIZED, isFirstTime).apply();
+    }
+
+    public String getZip(Context context) {
+        return getSharedPrefs(context).getString(KEY_USER_ZIP, "");
+    }
+
+    public void setZip(Context context, String zip) {
+        getSharedPrefs(context).edit().putString(KEY_USER_ZIP, zip).apply();
+    }
+
+    // Unfortunately, this has to be nullable because 0,0 is a valid lat,lng, so there's not really a sensical default value.
+    @Nullable
+    public Double getLat(Context context) {
+        // Use a long to store the bits, http://stackoverflow.com/questions/16319237/cant-put-double-sharedpreferences.
+        long bits = getSharedPrefs(context).getLong(KEY_LATITUDE, Long.MAX_VALUE);
+        if (bits == Long.MAX_VALUE) {
+            return null;
+        }
+        return Double.longBitsToDouble(bits);
+    }
+
+    public void setLat(Context context, Double lat) {
```

Comment:
Yes they should be, I figured it was better to store them as doubles so we didn't have to convert back and forth when using them outside of this class, like if we wanted to show them on a map.  However, since it looks like we're only using them as strings, to send to the api, it might be better to go back to storing them as strings.  Your call here.